get_genotype_vcf.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+# pylint: disable=import-error
+
+"""
+This script will
+
+- extract common variants
+- export as VCF (one per chrom)
+
+this will be used as input for the
+SAIGE-QTL association pipeline.
+"""
+
+# python modules
+from cpg_utils.hail_batch import (
+    dataset_path,
+    get_batch,
+    get_config,
+    init_batch,
+    output_path,
+)
+
+import click
+import hail as hl
+from hail.methods import export_vcf
+
+
+BIOHEART_TOB_VDS = dataset_path('vds/1-0.vds')
+BCFTOOLS_IMAGE = get_config()['images']['bcftools']
+
+# inputs:
+@click.option('--chromosomes', help=' eg chr22')
+@click.command()
+def main(chromosomes):
+    """
+    Run associaTR processing pipeline
+    """
+
+    init_batch()
+
+    vds_path = BIOHEART_TOB_VDS
+    vds = hl.vds.read_vds(vds_path)
+
+    for chromosome in chromosomes:
+
+        # consider only relevant chromosome
+        vds = hl.vds.filter_chromosomes(vds, keep=chromosome)
+
+        # split multiallelic loci
+        vds = hl.vds.split_multi(vds, filter_changed_loci=True)
+
+        # densify to matrix table object
+        mt = hl.vds.to_dense_mt(vds)
+
+        # filter out loci & variant QC
+        mt = mt.filter_rows(
+            ~(mt.was_split)  # biallelic (exclude multiallelic)
+            & (hl.len(mt.alleles) == 2)  # remove hom-ref
+            & (hl.is_snp(mt.alleles[0], mt.alleles[1]))  # SNPs (exclude indels)
+        )
+        mt = hl.variant_qc(mt)
+
+        # common variants only
+        cv_mt = mt.filter_rows(mt.variant_qc.AF[1] > 0.05)
+
+        # remove fields not in the VCF
+        cv_mt = cv_mt.drop('gvcf_info')
+
+        # export to plink common variants only
+        cv_vcf_path = output_path(f'{chromosome}_common_variants.vcf.bgz')
+        export_vcf(cv_mt, cv_vcf_path)
+
+        # add index file (.csi) using bcftools
+        vcf_input = get_batch().read_input(cv_vcf_path)
+        bcftools_job = get_batch().new_job(name='index vcf')
+        bcftools_job.image(BCFTOOLS_IMAGE)
+        bcftools_job.cpu(4)
+        bcftools_job.storage('15G')
+        bcftools_job.command(f"bcftools index -c {vcf_input} -o {cv_vcf_path}.csi")
+
```

Comment:
```suggestion
        bcftools_job.command(f"bcftools index -c {vcf_input} -o {bcftools_job.csi}")
        get_batch().write_output(bcftools_job.csi, f'{cv_vcf_path}.csi')
```

afaik bcftools won't be able to write directly to the GCP path, so you need to create a temp file then copy to GCP separately

Reply:
ah yes i see! Is that the syntax to get the job's output?? Just job name + .csi?

Reply:
every time you use `job.something` Hail reserves a name in the batch temp space for `something`, which can be accessed after to pass into other jobs or to copy into permanent storage. In the hail batch tutorials I'm pretty sure they use `job.output` each time, but any name will work.